src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -489,11 +489,7 @@ ReceiveShardNameAndSizeResults(List *connectionList, Tuplestorestate *tupleStore
 			memset(isNulls, false, sizeof(isNulls));
 
 			/* format is [0] shard id, [1] shard name, [2] size */
-			char *tableName = PQgetvalue(result, rowIndex, 1);
-			Datum resultStringDatum = CStringGetDatum(tableName);
-			Datum textDatum = DirectFunctionCall1(textin, resultStringDatum);
-
-			values[0] = textDatum;
+			values[0] = ParseIntField(result, rowIndex, 0);
```

Comment:
Afaict there's now no function that actually reads the the shard name column that the query returns. So I think we can remove that from the query we send internally. See `AppendShardSizeQuery` for where the query is created.